pkg/stats/view.go

```diff
@@ -0,0 +1,296 @@
+package stats
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/fastly/go-fastly/fastly"
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+	"github.com/mitchellh/mapstructure"
+)
+
+// Maximum number of time slots to remember (1 sec per spot)
+const maxSlots = 60
+
+type View struct {
+	Events  <-chan ui.Event
+	Grid    *ui.Grid
+	Service string
+
+	// Header
+	Header *widgets.Paragraph
+
+	// Top Row
+	Hits     *widgets.Paragraph
+	HitTime  *widgets.Paragraph
+	Misses   *widgets.Paragraph
+	HitRatio *widgets.PieChart
+	MissTime *widgets.Paragraph
+	Requests *widgets.Paragraph
+	Errors   *widgets.Paragraph
+
+	// Second Row
+	GlobalLabel *widgets.Paragraph
+	Global      *widgets.BarChart
+	GlobalData  []float64
+
+	// Third Row
+	ReqLabel       *widgets.Paragraph
+	ReqChart       *widgets.StackedBarChart
+	ReqData        [][]float64
+	BandwidthLabel *widgets.Paragraph
+	BandwidthGroup *widgets.SparklineGroup
+	BandwidthChart *widgets.Sparkline
+	BandwidthData  []float64
+
+	// Fourth Row
+	ErrLabel *widgets.Paragraph
+	ErrGroup *widgets.SparklineGroup
+	ErrChart *widgets.Sparkline
+	ErrData  []float64
+	HitLabel *widgets.Paragraph
+	HitGroup *widgets.SparklineGroup
+	HitChart *widgets.Sparkline
+	HitData  []float64
+
+	// Fifth Row
+	IoLabel      *widgets.Paragraph
+	IoGroup      *widgets.SparklineGroup
+	IoChart      *widgets.Sparkline
+	IoData       []float64
+	LoggingLabel *widgets.Paragraph
+	LoggingGroup *widgets.SparklineGroup
+	LoggingChart *widgets.Sparkline
+	LoggingData  []float64
+}
+
+func NewView(service string) (*View, error) {
+	var view = View{}
+	view.Service = service
+
+	if err := ui.Init(); err != nil {
```

Comment:
Calling `ui.Init` here seems orthogonal to creating a new View, and if this doesn't Init() then NewView doesn't need to return an error at all: that's useful because it signals to other code that this can never fail.

Depending on what happens in `ui.Init` it may also unnecessarily prevent you from calling NewView twice.